test/BloomFilterTest.cpp

```diff
@@ -16,19 +16,51 @@
 
 #include <iostream>
 #include <uuid/uuid.h>
+#include <stdio.h>
+#include <unistd.h>
 #include "BloomFilter.hpp"
 
-#define CATCH_CONFIG_MAIN
-
+#define CATCH_CONFIG_RUNNER
 #include "catch.hpp"
 
-
 using namespace std;
 
 static const unsigned int FILTER_ELEMENT_COUNT = 5000;
 static const unsigned int ADDITIONAL_TEST_DATA_ELEMENT_COUNT = 5000;
+
 static const double TARGET_ERROR_RATE = 0.001;
-static const double ACCEPTABLE_ERROR_RATE = TARGET_ERROR_RATE * 2;
+static const double ACCEPTABLE_ERROR_RATE = TARGET_ERROR_RATE * 2.5;
+
+static void setup();
+static set<string> createRandomStrings(unsigned int count);
+static bool testBloomFilter(BloomFilter &bloomFilter, set<string> &bloomData, set<string> &testData);
+static void writeStringsToFile(const set<string> &strings, const string &fileName);
+static set<string> readStringsFromFile(const string &fileName);
+
+// Global strings for creating and testing the bloom filters
+static set<string> gBloomData;
+static set<string> gTestData;
+
+int main( int argc, char* argv[] ) {
+  // global setup
+  setup();
+  // run tests
+  int result = Catch::Session().run( argc, argv );
+  // global clean-up - none
+  return result;
+}
+
+static void setup() {
+    // Create the global bloomfilter data
+    BloomFilter bloomfilter1(FILTER_ELEMENT_COUNT, TARGET_ERROR_RATE);
+    gBloomData = createRandomStrings(FILTER_ELEMENT_COUNT);
+    gTestData = createRandomStrings(ADDITIONAL_TEST_DATA_ELEMENT_COUNT);
+    gTestData.insert(gBloomData.begin(), gBloomData.end());
+
+    // TODO use filesystem::temp_directory_path() to get platform independent tmp directory
```

Comment:
Resolve todos across the PR: We only use todos as reminders during dev, by the time a feature hits prod the todos should be resolved.